support-frontend/assets/components/checkoutSupportOnce/checkoutSupportOnce.tsx

```diff
@@ -0,0 +1,114 @@
+import { css } from '@emotion/react';
+import {
+	brand,
+	from,
+	headline,
+	neutral,
+	space,
+	textSans,
+	until,
+} from '@guardian/source-foundations';
+import type { ContributionType } from 'helpers/contributions';
+import { CheckoutSupportOnceButton } from './checkoutSupportOnceButton';
+
+const container = css`
+	display: flex;
+	flex-direction: column;
+	margin: 10px ${space[3]}px ${space[4]}px;
+
+	${from.desktop} {
+		margin: ${space[5]}px ${space[6]}px;
+	}
+`;
+const top = css`
+	display: flex;
+	flex-direction: row;
+	margin-left: 0px;
+
+	${from.desktop} {
+		margin-left: 2px;
+	}
+`;
+const bottom = css`
+	display: flex;
+	flex-direction: row;
+	justify-content: space-between;
+
+	${from.desktop} {
+		margin-left: 0px;
+	}
+`;
+const heading = (color: string) => css`
+	${headline.xxsmall({ fontWeight: 'bold', lineHeight: 'regular' })};
+	color: ${color};
+
+	${from.desktop} {
+		${headline.xsmall({ fontWeight: 'bold', lineHeight: 'tight' })};
+	}
+`;
+const para = css`
+	text-align: left;
+	${textSans.small({ lineHeight: 'regular' })};
+
+	${from.desktop} {
+		${textSans.medium({ lineHeight: 'regular' })};
+	}
+`;
+const box = css`
+	:not(:last-child) {
+		${until.tablet} {
+			margin-bottom: ${space[2]}px;
+		}
+	}
+	&:hover {
+		cursor: pointer;
+	}
+`;
+const mainStyles = css`
+	display: block;
+	overflow: hidden;
+	background-color: ${neutral[100]};
+	color: ${neutral[7]};
+	border: 1px solid ${neutral[86]};
+	border-radius: ${space[3]}px;
+
+	:not(:last-child) {
+		margin-bottom: ${space[3]}px;
+
+		${from.mobileLandscape} {
+			margin-bottom: ${space[4]}px;
+		}
+	}
+`;
+
+export type CheckoutSupportOnceProps = {
+	contributionType: ContributionType;
+	supportOnceTitle: string;
+	supportOnceParagraph: string;
+	onSupportOnceClick: () => void;
+};
+
+export function CheckoutSupportOnce({
+	contributionType,
+	supportOnceTitle,
+	supportOnceParagraph,
+	onSupportOnceClick,
+}: CheckoutSupportOnceProps): JSX.Element | null {
+	if (contributionType !== 'ONE_OFF') {
+		return (
+			<a onClick={onSupportOnceClick} css={[mainStyles, box]}>
```

Comment:
Having a `<button>` inside an `<a>`, which both have the same click handler, is a huge anti-pattern. Do not do this. I'm not sure that it's a good idea to contain the entire component inside an anchor- I would want to run through it with VoiceOver to check but this is potentially going to make this link confusing and/or meaningless to anyone using the rotor menu to look for links on the page, and confusing to interact with for anyone using tab navigation.

The CSS Tricks article I linked on the previous PR suggests using an anchor (or button) _inside_ the 'clickable card', and then using some hacky positioning to extend an interactive area across the whole containing component. Please have a read of it in full: https://css-tricks.com/creating-animated-clickable-cards-with-the-has-relational-pseudo-class/ While I still don't think we should be doing this at all, I think that would be the best approach if we are committed to this component- it keeps the HTML structure as sensible as possible for assistive technologies while still allowing the desired interaction.

Reply:
re: <button> inside an <a>  am sorry I didn't know it was a huge no-no. Am compiling a list of 'don't do' rules in my notes for future reference, this is in there. Ok, the button has been removed from the component inside the anchor at least now.

Have gone through the article and think the approach is fundamentally different from the nudge. Agree there is much more complexity in building this the right way as a standalone clickable component and testing it. 

From reading this closer, would guess a component useEffect would be needed so the JS in the article can convert mouse pressing within any hypertext labeled 'card's to 'a' click (which is inside the card). If working, would the approach still be acceptable as its mentioned as being a bit hacky?

I don't think I can complete this alone, then test it and in time before the end of the week. 

Perhaps I close this card as it stands, without putting it into prod, and we refer to it in future if needed? Then create a new 'investigate clickable component' trello card to do as suggested above (and test plus decide if acceptable). Followed by another trello card to integrate it into supportJustOnce (perhaps referring to parts of this PR if necessary). Would that work?

Reply:
I think rather than trying to compile a set of rules, the most helpful thing is usually to take a step back from being deep in the weeds of the problem and have a think about the actual HTML structure you're creating, outside of the design and the business needs. Good semantic HTML really, really does get you most of the way there with accessibility (which is one of the reasons why if we're asked to build something that doesn't have an obvious HTML implementation it should give us pause).

In this case, try thinking about what an anchor tag and a button actually _are_, and what we're trying to communicate by using them. If for some reason our styles failed to work on this page, what would it look like? Would that kind of interface look or feel odd to use? If you were reading that raw HTML, would it seem a bit odd to you?

Don't feel like you have to rush, or that you have to jump immediately to follow mine or anyone else's suggestions without fully comprehending the problem for yourself. It's _always_ better to take a little bit more time to have more confidence in your code. If we have to push the deadline a bit then so be it.

Reply:
Ok, have created a clickable card component that produces a click event upon mouse pressing any html element within a div labelled 'card' (by listening & timing mouseup/mousedown events). IMoe generic and its based upon  https://css-tricks.com/creating-animated-clickable-cards-with-the-has-relational-pseudo-class/ Obviously subject to tomorrows meeting over accesibility and whether we should do this at all but at least its there now.